ci/workflows/pull-request.yml

```diff
@@ -42,6 +42,16 @@ steps:
             - CODECOV_TOKEN
     timeout_in_minutes: 15
 
+  - label: "deterministic simulation test"
+    command: "ci/scripts/deterministic-simulation-test.sh"
+    plugins:
+      - docker-compose#v3.9.0:
+          run: rw-build-env
+          config: ci/docker-compose.yml
+          mount-buildkite-agent: true
+    timeout_in_minutes: 5
+    soft_fail: true
```

Comment:
Can you tell me why soft_fail is used here?

Reply:
Deterministic test is not stable enough. It will fail randomly. We will try to make it more stable in the future.

Reply:
OK, LGTM!